app/src/androidTest/java/com/google/jetpackcamera/utils/UiTestUtil.kt

```diff
@@ -77,3 +90,129 @@ fun ComposeTestRule.getCurrentLensFacing(): LensFacing {
         }
     }
 }
+
+/**
+ * Rule that you to specify test methods that will have permissions granted prior to starting
+ *
+ * @param permissions the permissions to be granted
+ * @param targetTestNames the names of the tests that this rule will apply to
+ */
+class IndividualTestGrantPermissionRule(
+    private val permissions: Array<String>,
+    private val targetTestNames: Array<String>
+) :
+    TestRule {
+    private lateinit var wrappedRule: GrantPermissionRule
+
+    override fun apply(base: Statement, description: Description): Statement {
+        for (targetName in targetTestNames) {
+            if (description.methodName == targetName) {
+                wrappedRule = GrantPermissionRule.grant(*permissions)
+                return wrappedRule.apply(base, description)
+            }
+        }
+        // If no match, return the base statement without granting permissions
+        return base
+    }
+}
+
+// functions for interacting with system permission dialog
+@SdkSuppress(minSdkVersion = 30)
```

Comment:
nit: `@SdkSuppress` is only needed on methods annotation with `@Test`. This one isn't needed.